src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -761,7 +751,7 @@ ss::future<> service::set_metadata(job_id_t job_id) {
       job_id);
 
     try {
-        co_await write_file(process_output_file.native(), std::move(po_buf));
+        co_await write_fully(process_output_file, std::move(po_buf));
```

Comment:
> The file was being flushed async (why?), and reading the file was racing
with the truncation after the direct I/O write.

I'm not following this. From what I can tell, `write_file` was synchronously flushing (last line in the function). With respect to the race, what's the read path you were referring to?

Reply:
> last line in the function 

Ah that is not flushing the file but the stream!

Reply:
Oh wait maybe the stream flush does flush the file... Hrm, the race must be elsewhere

Reply:
@nvartolomei knows for sure i think he filed a ticket recently in seastar to change this behavior

Reply:
Okay with Mike's help I tracked down the proper fix: https://github.com/redpanda-data/redpanda/pull/26060